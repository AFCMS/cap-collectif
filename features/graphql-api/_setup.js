/* eslint-env jest */
const util = require('util');
const exec = util.promisify(require('child_process').exec);

const env = process.env.CI ? 'ci' : 'local';

let isFirstRun = true;

beforeAll(async () => {
  if (isFirstRun) {
    console.log('Saving database…');
    console.time('save_db');
    const { stderrDb } = await exec('fab ' + env + '.qa.save_db');

    if (stderrDb) {
      console.error(`error: ${stderrDb}`);
    }
    console.log('Successfully saved database');
    console.timeEnd('save_db');

    console.log('Writing ElasticSearch snapshot…');
    console.time('save_es');
    const { stderr } = await exec('fab ' + env + '.qa.save_es_snapshot');

    if (stderr) {
      console.error(`error: ${stderr}`);
    }
    console.log('Successfully saved ElasticSearch snapshot');
    console.timeEnd('save_es');
  }
  isFirstRun = false;
});

afterEach(async () => {
  console.log('Purging RabbitMQ…');
  console.time('purge_mq');
  await exec('fab ' + env + '.qa.purge_rabbitmq');
  console.log('Successfully purged RabbitMQ !');
  console.timeEnd('purge_mq');
});

afterAll(async () => {
  console.log('Restoring database…');
  console.time('restore_db');
  const { stderrDb } = await exec('fab ' + env + '.qa.restore_db');

  if (stderrDb) {
    console.error(`error: ${stderrDb}`);
  }
  console.log('Successfully restored database !');
  console.timeEnd('restore_db');

  console.log('Restoring ElasticSearch snapshot…');
  console.time('restore_es');
  const { stderrEs } = await exec('fab ' + env + '.qa.restore_es_snapshot');
  if (stderrEs) {
    console.error(`error: ${stderrEs}`);
  }
  console.log('Successfully restored ES !');
  console.timeEnd('restore_es');
});
