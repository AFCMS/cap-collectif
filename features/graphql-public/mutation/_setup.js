/* eslint-env jest */
const util = require('util');
const exec = util.promisify(require('child_process').exec);

beforeAll(async () => {
  console.log('Writing ElasticSearch snapshot...');
  const { stderr } = await exec('fab local.qa.save_es_snapshot');

  if (stderr) {
    console.error(`error: ${stderr}`);
  }
  console.log('Successfully saved ElasticSearch snapshot');
});

afterAll(async () => {
  console.log('Restoring database...');
  const { stderrDb } = await exec('fab local.qa.restore_db');

  if (stderrDb) {
    console.error(`error: ${stderrDb}`);
  }
  console.log('Successfully restored database');

  console.log('Restoring ElasticSearch snapshot.');
  const { stderrEs } = await exec('fab local.qa.restore_es_snapshot');

  if (stderrEs) {
    console.error(`error: ${stderrEs}`);
  }
});
