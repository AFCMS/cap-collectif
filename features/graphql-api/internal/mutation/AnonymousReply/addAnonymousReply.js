/* eslint-env jest */
import '../../../_setup';

const AddAnonymousReplyMutation = /* GraphQL*/ `
    mutation AddAnonymousReplyMutation($input: AddAnonymousReplyInput!) {
        addAnonymousReply(input: $input) {
            reply {
                __typename
                id
                participantEmail
            }
            questionnaire {
                id
            }
            token
        }
    }
`;

const input = {
  "questionnaireId": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
  "responses": [
    {
      "value": "abc",
      "question": "UXVlc3Rpb246MTQwMw=="
    }
  ],
  "participantEmail": null
};

const propertyMatchers = {
  addAnonymousReply: {
    reply: {
      id: expect.any(String)
    },
    token: expect.any(String)
  },
}

describe('mutations.addAnonymousReply', () => {
  it('should create an anonymous reply', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input,
      },
      'internal'
    );

    expect(response).toMatchSnapshot(propertyMatchers);
  });

  it('should create an anonymous reply with participantEmail', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input: {
          ...input,
          participantEmail: 'abc@cap-collectif.com'
        }
      },
      'internal'
    );

    expect(response).toMatchSnapshot(propertyMatchers);
  });
});
