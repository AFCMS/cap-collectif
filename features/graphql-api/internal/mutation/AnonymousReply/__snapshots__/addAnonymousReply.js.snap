// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.addAnonymousReply should create an anonymous reply 1`] = `
Object {
  "addAnonymousReply": Object {
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": Object {
      "__typename": "AnonymousReply",
      "id": Any<String>,
      "participantEmail": null,
    },
    "token": Any<String>,
  },
}
`;

exports[`mutations.addAnonymousReply should create an anonymous reply with participantEmail 1`] = `
Object {
  "addAnonymousReply": Object {
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": Object {
      "__typename": "AnonymousReply",
      "id": Any<String>,
      "participantEmail": "abc@cap-collectif.com",
    },
    "token": Any<String>,
  },
}
`;
